pnpm/src/main.ts

```diff
@@ -325,3 +332,27 @@ function printError (message: string, hint?: string): void {
     console.log(hint)
   }
 }
+
+function checkPackageManager (pm: WantedPackageManager, config: Config): void {
+  if (pm.name && pm.name !== 'pnpm') {
```

Comment:
Since `WantedPackageManager.name` is not optional according to the type definition of `WantedPackageManager` in `@pnpm/config`, the only way `pm.name` could be falsy here is an empty string. I think we should have an error for when `pm.name` being empty string.

If instead the parser (that converts `packageManager` string to `WantedPackageManager` object) already errors on empty `name`, `pm.name &&` should be removed, leaving only `if (pm.name !== 'pnpm')`.

```suggestion
  if (pm.name !== 'pnpm') {
```